db/migrate/20230502083009_create_shoutouts.rb

```diff
@@ -0,0 +1,14 @@
+class CreateShoutouts < ActiveRecord::Migration[7.0]
+  def change
+    create_table :shoutouts do |t|
+      t.text :rich_text, null: false
+      t.string :recipients
+      t.references :user, null: false, foreign_key: true
+      t.references :time_period, null: false, foreign_key: true
+      t.bigint :digest, null: false, unique: true
```

Comment:
Please use model validations to prevent duplication (for example combinations of `rich_text` + `user_id` + `time_period`) unless:

 - This is known RoR pattern (please send me some links of articles written about this approach being better than the traditional), and:
 - Your team votes unanimously to adopt this pattern across the entire project and it's going to be implemented everywhere.
 
Why traditional patterns trumps custom solutions, even if the custom solution is more efficient:

1. Open source participation: Developers from the outside considering building upon our work will be confused as to why something is done in a strange way that they aren't used to. Unless an obvious reason is apparent they will be reluctant to touch the project.

2. Internal training: Every new developer who joins our team from now until the end of time will ask the same thing and we'll have to answer again and again, and/or provide documentation for why duplication is dealt with one way here and everywhere else it's done another way.

Reply:
Roger, this method was created by me, because I did not find an answer on the Internet how to avoid duplication by having a text field and a few additional ones. But thanks to Svetlana, who showed me how to do it at the model level much easier, so it will be overwritten and there will be no questions from other developers